packages/graphqlgen/src/tests/typescript/__snapshots__/basic.test.ts.snap

```diff
@@ -0,0 +1,342 @@
+    height: (parent: Dimensions) => parent.height
+  };
+
+  export type WidthResolver = (
+    parent: Dimensions,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => number | Promise<number>;
+
+  export type HeightResolver = (
+    parent: Dimensions,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => number | Promise<number>;
+
+  export interface Type {
+    width: (
+      parent: Dimensions,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => number | Promise<number>;
+
+    height: (
+      parent: Dimensions,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => number | Promise<number>;
+  }
+}
+
+export namespace ImageResolvers {
+  export const defaultResolvers = {
+    dimensions: (parent: Image) => parent.dimensions
+  };
+
+  export type IdResolver = (
+    parent: Image,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => string | Promise<string>;
+
+  export type UrlResolver = (
+    parent: Image,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => string | Promise<string>;
+
+  export type DimensionsResolver = (
+    parent: Image,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => Dimensions | Promise<Dimensions>;
+
+  export interface Type {
+    id: (
+      parent: Image,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => string | Promise<string>;
+
+    url: (
+      parent: Image,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => string | Promise<string>;
+
+    dimensions: (
+      parent: Image,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => Dimensions | Promise<Dimensions>;
+  }
+}
+
+export namespace VideoResolvers {
+  export const defaultResolvers = {};
+
+  export type IdResolver = (
+    parent: Video,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => string | Promise<string>;
+
+  export type UrlResolver = (
+    parent: Video,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => string | Promise<string>;
+
+  export type DurationResolver = (
+    parent: Video,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => number | Promise<number>;
+
+  export interface Type {
+    id: (
+      parent: Video,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => string | Promise<string>;
+
+    url: (
+      parent: Video,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => string | Promise<string>;
+
+    duration: (
+      parent: Video,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => number | Promise<number>;
+  }
+}
+
+export interface Resolvers {
+  Query: QueryResolvers.Type<T>;
+  Dimensions: DimensionsResolvers.Type<T>;
+  Image: ImageResolvers.Type<T>;
+  Video: VideoResolvers.Type<T>;
+  Media?: MediaResolvers.Type<T>;
```

Comment:
I currently marked unions and interfaces as optional, but I think this might not be right. Thoughts?